application/src/main/java/org/togetherjava/tjbot/commands/filesharing/FileSharingMessageListener.java

```diff
@@ -0,0 +1,178 @@
+        GistFiles files = new GistFiles(filesAsJson);
+        GistRequest request = new GistRequest(event.getAuthor().getName(), false, files);
+        String url = uploadToGist(request);
+        sendResponse(event, url);
+    }
+
+    private @NotNull String readAttachment(@NotNull InputStream stream) {
+        try {
+            return new String(stream.readAllBytes(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private @NotNull String createFileName(@NotNull Message.Attachment attachment) {
+        String fileName = attachment.getFileName();
+        String fileExtension = attachment.getFileExtension();
+
+        if (fileExtension == null || fileExtension.equals("txt")) {
+            fileExtension = "java";
+        } else if (fileExtension.equals("fxml")) {
+            fileExtension = "xml";
+        }
+
+        fileName = fileName.substring(0, fileName.lastIndexOf(".")) + "." + fileExtension;
+        return fileName;
+    }
+
+    private @NotNull String uploadToGist(@NotNull GistRequest jsonRequest) {
+        String body;
+        try {
+            body = JSON.writeValueAsString(jsonRequest);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Couldn't parse json request!", e);
+        }
+
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(SHARE_API))
+            .header("Accept", "application/json")
+            .header("Authorization", "token " + gistApiKey)
+            .POST(HttpRequest.BodyPublishers.ofString(body))
+            .build();
+
+        HttpResponse<String> apiResponse;
+        try {
+            apiResponse = CLIENT.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
+        int statusCode = apiResponse.statusCode();
+
+        if (statusCode < HttpURLConnection.HTTP_OK
+                || statusCode >= HttpURLConnection.HTTP_MULT_CHOICE) {
+            throw new IllegalStateException("Gist API unexpected response: " + apiResponse.body());
+        }
+
+        GistResponse gistResponse;
+        try {
+            gistResponse = JSON.readValue(apiResponse.body(), GistResponse.class);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Couldn't parse response!", e);
+        }
+        return gistResponse.getHtmlUrl();
+    }
+
+    private void sendResponse(@NotNull MessageReceivedEvent event, @NotNull String url) {
+        Message message = event.getMessage();
+        String replyContent =
+                "I uploaded your file(s) as gist, it is much easier to read for everyone like that, especially for mobile users";
+
+        message.reply(replyContent).setActionRow(Button.of(ButtonStyle.LINK, url, "gist")).queue();
```

Comment:
Suggestion:
https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/interactions/components/buttons/Button.html#link(java.lang.String,java.lang.String)

`Button` class have a specific method to create a link style button

Reply:
actually just do it

it makes the code look better

```java
.setActionRow(Button.link(url, "Gist"))
```

Reply:
oh didn't know that this exists, gonna change that